File: PreUtil/src/output/c/medium/SubarraySumsDivisibleByK.c
/**https://leetcode.com/problems/subarray-sums-divisible-by-k/ */
//Given an integer array nums and an integer k, return the number of non-empty subarrays that have a sum divisible by k.
//A subarray is a contiguous part of an array.


int subarraysDivByK(int* nums, int numsSize, int k){
    int* count = (int*)malloc(sizeof(int) * k);
    for(int i = 0; i < k; i++){
        count[i] = 0;
    }
    int sum = 0;
    for(int i = 0; i < numsSize; i++){
        sum += nums[i];
        count[((sum % k) + k) % k]++;
    }
    int result = 0;
    for(int i = 0; i < k; i++){
        result += count[i] * (count[i] - 1) / 2;
    }
    return result + count[0];

}

    

File: Output/javascript/medium/SubarraySumsDivisibleByK.js
/**https://leetcode.com/problems/subarray-sums-divisible-by-k/ */
//Given an integer array nums and an integer k, return the number of non-empty subarrays that have a sum divisible by k.
//A subarray is a contiguous part of an array.


/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraysDivByK = function(nums, k) {
    let count = new Array(k).fill(0);
    count[0] = 1;
    let sum = 0;
    let ans = 0;
    for(let i = 0; i < nums.length; i++){
        sum += nums[i];
        let mod = ((sum % k) + k) % k;
        ans += count[mod];
        count[mod]++;
    }
    return ans;
    
};

    